File: solutions/cpp/leap/leap.cpp
#include "leap.h"

namespace leap {

    bool is_divisible_by(int number, int divisor) {
        return (number % divisor) == 0;
    }

    bool is_leap_year(int year) {
        return is_divisible_by(year, 4) && (!is_divisible_by(year, 100) || is_divisible_by(year, 400));
    }
}  // namespace leap

File: solutions/java/leap/src/main/java/Leap.java
class Leap {

    boolean isLeapYear(int year) {
        return isDivisibleBy(year, 4) && (!isDivisibleBy(year, 100) || isDivisibleBy(year, 400));
    }

    private boolean isDivisibleBy(int dividend, int divisor) {
        return dividend % divisor == 0;
    }
}
